pkg/kubelet/cm/cpumanager/topology/topology_test.go

```diff
@@ -503,6 +502,36 @@ func Test_Discover(t *testing.T) {
 			want:    &CPUTopology{},
 			wantErr: true,
 		},
+		{
+			name: "OneSocketHT second thread lowest",
+			machineInfo: cadvisorapi.MachineInfo{
+				NumCores:   4,
+				NumSockets: 1,
+				Topology: []cadvisorapi.Node{
+					{Id: 0,
+						Cores: []cadvisorapi.Core{
+							{SocketID: 0, Id: 0, Threads: []int{0, 1}},
+							{SocketID: 0, Id: 1, Threads: []int{1, 2}},
+							{SocketID: 0, Id: 2, Threads: []int{2, 3}},
+							{SocketID: 0, Id: 3, Threads: []int{3, 2}}, // Corner case - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/topology/topology.go#L268-L273
```

Comment:
in general we try to cover case which are corner, but realistic, stemming from actual HW topologies implemented in silicon. Is something we can observe in any supported CPU?

Reply:
resolved in 512ca4b9888c303b1b0aef5469293a099bc4cd9b